packages/react-ui/components/Modal/Modal.tsx

```diff
@@ -199,27 +201,59 @@ export class Modal extends React.Component<ModalProps, ModalState> {
-                })}
-                style={containerStyle}
-                data-tid="modal-content"
-              >
-                <div className={styles.window(this.theme)} style={style}>
-                  <ResizeDetector onResize={this.handleResize}>
-                    <FocusLock disabled={this.props.disableFocusLock} autoFocus={false}>
-                      {!hasHeader && !this.props.noClose ? (
-                        <ZIndex priority={'ModalCross'} className={styles.closeWrapper(this.theme)}>
-                          <ModalClose requestClose={this.requestClose} disableClose={this.props.disableClose} />
-                        </ZIndex>
-                      ) : null}
-                      <ModalContext.Provider value={modalContextProps}>{this.props.children}</ModalContext.Provider>
-                    </FocusLock>
-                  </ResizeDetector>
-                </div>
-              </div>
+              <ResponsiveLayout>
+                {({ isMobile }) => {
+                  return (
+                    <div
+                      className={cx({
+                        [styles.centerContainer(this.theme)]: true,
+                        [styles.mobileCenterContainer()]: isMobile,
+                        [styles.alignTop()]: Boolean(this.props.alignTop),
+                      })}
+                      style={isMobile ? undefined : containerStyle}
+                      data-tid="modal-content"
+                    >
+                      <div
+                        className={cx({ [styles.window(this.theme)]: true, [styles.mobileWindow()]: isMobile })}
+                        style={isMobile ? undefined : style}
+                      >
+                        <ResizeDetector onResize={this.handleResize} fullHeight={isMobile}>
+                          <FocusLock
+                            disabled={this.props.disableFocusLock}
+                            autoFocus={false}
+                            className={cx({ [styles.columnFlexContainer()]: isMobile }, 'focus-lock-container')}
+                          >
+                            {!hasHeader && !this.props.noClose && (
+                              <ZIndex
+                                priority={'ModalCross'}
+                                className={cx({
+                                  [styles.closeWrapper(this.theme)]: true,
+                                  [styles.mobileCloseWrapper(this.theme)]: isMobile,
+                                })}
+                              >
+                                <ModalClose
+                                  className={
+                                    isMobile
+                                      ? cx({
+                                          [styles.mobileCloseWithoutHeader()]: !this.state.hasHeader,
+                                        })
+                                      : undefined
+                                  }
```

Comment:
Без веток было бы получше.

```suggestion
                                  className={cx({
                                    [styles.mobileCloseWithoutHeader()]: isMobile && !this.state.hasHeader,
                                  })}
```